### PR TITLE
[Auditbeat] Use a separate netlink socket for control to avoid data congestion.

### PR DESCRIPTION
**Pull request by nicholasberlin**
## Proposed commit message
Use a separate socket for GetStatus.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Run this [program](https://gist.github.com/adriansr/16524e5b1242ce5cf59ca623a59a9ddb#file-bomb-go) while starting Auditbeat. In a separate terminal run `while true; do ss -f netlink | grep auditbeat; done`

You will see the auditbeat netlink sockets disappear but Auditbeat will continue to run.
Data will stop flowing to Elasticsearch, and an error message will be pushed Elasticsearch, which will be similar to this:
image

With this PR's patch, the netlink sockets will remain, data will flow to Elasticsearch, and no error message will be pushed.

### CODE REVIEW

File: auditbeat/module/auditd/audit_linux.go [hunk 1]
```diff
@@ -164,14 +164,30 @@ func closeAuditClient(client *libaudit.AuditClient, log *logp.Logger) {
 	}
 }
 
+func getStatus() (*libaudit.AuditStatus, error) {
+	client, err := libaudit.NewAuditClient(nil)
+	defer func() {
+		if client != nil {
+			client.Close()
```
Comment (fearful-symmetry) [inline auditbeat/module/auditd/audit_linux.go:171]: I assume that if we move the `defer` statement to after the error check, we don't need the nil check, and we can just do `defer client.close()`

File: auditbeat/module/auditd/audit_linux.go [hunk 2]
```diff
@@ -164,14 +164,30 @@ func closeAuditClient(client *libaudit.AuditClient, log *logp.Logger) {
...
+		}
+	}()
+
+	if err != nil {
+		return nil, err
```
Comment (fearful-symmetry) [inline auditbeat/module/auditd/audit_linux.go:176]: nit: can we get a `fmt.Errorf()` here?

**Review by fearful-symmetry — COMMENTED** [review]
just some small nits

File: auditbeat/module/auditd/audit_linux.go (hunk 1 above)
Comment (nicholasberlin) [reply auditbeat/module/auditd/audit_linux.go:171]: Ah, makes sense. Thanks.

Comment (nicholasberlin) [reply auditbeat/module/auditd/audit_linux.go]: a6b5df3

File: auditbeat/module/auditd/audit_linux.go (hunk 2 above)
Comment (nicholasberlin) [reply auditbeat/module/auditd/audit_linux.go]: a6b5df3

**Comment by nicholasberlin** [comment]
@fearful-symmetry I re-worked the whole thing, please have another look. Thanks.

**Comment by nicholasberlin** [comment]
> we need to use value, ok := interface.(type)

Thanks! Will get into it tomorrow.

**Comment by nicholasberlin** [comment]
@fearful-symmetry ready for review now. Thanks again.

File: auditbeat/module/auditd/audit_linux.go [hunk 3]
```diff
@@ -168,10 +175,13 @@ func closeAuditClient(client *libaudit.AuditClient, log *logp.Logger) {
 // kernel until the reporter's done channel is closed.
 func (ms *MetricSet) Run(reporter mb.PushReporterV2) {
 	defer closeAuditClient(ms.client, ms.log)
+	defer ms.control.Close()
```
Comment (andrewkroh) [inline auditbeat/module/auditd/audit_linux.go:178]: Do you think this should be using the same `closeAuditClient()` to close `ms.control`? Not sure of the history that necessitated closeAuditClient.

Comment (nicholasberlin) [reply auditbeat/module/auditd/audit_linux.go:178]: No, it doesn't need `closeAuditClient`. The [`Close`](https://github.com/elastic/go-libaudit/blob/main/audit.go#L425-L446) function of `AuditClient` only sends a message before closing if [`SetPID`](https://github.com/elastic/go-libaudit/blob/main/audit.go#L310-L320) has been called. Which it is hasn't for `ms.control`. 

From what I can glean from the [deadlock fix](https://github.com/elastic/beats/pull/26032), the message being sent was sometimes being dropped when the netlink socket was overwhelmed.  This led to a deadlock. To remedy the problem, a go routine was started to drain the socket in parallel to the message being sent.

Seems like calling `closeAuditClient` is harmless though. Would you like me to change it?

Comment (andrewkroh) [reply auditbeat/module/auditd/audit_linux.go:178]: No, thanks for investigating and explaining.

**Review by andrewkroh — APPROVED** [review]

**Review by fearful-symmetry — APPROVED** [review]

**Pull request merged**